MainActivity RecyclerView now populates with RobotView Items

aayush-fadia opened the pull request:
Screenshot
![Screenshot](https://i.ibb.co/WGT7CQ7/main-recycler.png)

aayush-fadia committed:
MainActivity RecyclerView now populates with RobotView Items

aayush-fadia commented:
Changed the isFavourite variable in the RobotViewData class to a Boolean.
All robots that are favoured come out on top of the RecyclerView.
Used some dummy RobotViewData items to test, and get the screenshot.
Am I doing this right?

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainActivity.kt:
```diff
@@ -1,12 +1,101 @@
+
+    /*Gets Dummy List of Robots to test RecyclerView Population*/
+    fun getDummyRobotList(): List<MainActivityViewInterface.RobotViewData> {
+        val ROBOTS_TO_DISPLAY = List<MainActivityViewInterface.RobotViewData>(5) {
```
Change variable name to robotsToDisplay. Maintain naming consistency
```suggestion
        val robotList = List<MainActivityViewInterface.RobotViewData>(5) {
```

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainActivity.kt:
```diff
@@ -1,12 +1,101 @@
+    val uuid = robot.UUID
+
+    class ViewHolder(view: View) : FastAdapter.ViewHolder<RobotItem>(view) {
+        var name_textView = view.findViewById<TextView>(R.id.robot_card_title)
```
Naming consistency use camelCase

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainActivity.kt:
```diff
@@ -1,12 +1,101 @@
+    }
+
+    /*Gets Dummy List of Robots to test RecyclerView Population*/
+    fun getDummyRobotList(): List<MainActivityViewInterface.RobotViewData> {
```
Don't want this here. Instead create a dummy presenter class to mock this test the view class.

aayush-fadia replied:
Done.

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainActivity.kt:
```diff
@@ -1,12 +1,101 @@
+    }
+}
+
+open class RobotItem(robot: MainActivityViewInterface.RobotViewData) :
```
Why is this an open class? We dont need to extend this class mark it as closed
```suggestion
class RobotItem(robot: MainActivityViewInterface.RobotViewData) :
```

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainActivity.kt:
```diff
@@ -1,12 +1,101 @@
+import com.mikepenz.fastadapter.items.AbstractItem
 
-class MainActivity : AppCompatActivity() {
+val Boolean.int
```
Orphan function get rid of this

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainActivity.kt:
```diff
@@ -1,12 +1,101 @@
+                    RobotItem(it)
+                }.toList()
+        )
+        Toast.makeText(this, "Here!", Toast.LENGTH_SHORT).show()
```
Remove toast msg. Use debugger instead to test your code at run time
```suggestion
```

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainActivity.kt:
```diff
@@ -1,12 +1,101 @@
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        displayRobotList(getDummyRobotList())
```
Test harness in production class. getDummy robot must be a presenter class mockup for testing

kamatajinkya reviewed (changes requested):
Test harness must not be a part of production class

aayush-fadia committed:
Make class closed

Co-authored-by: Ajinkya Kamat <kamat.ajinkya@gmail.com>

aayush-fadia committed:
Make name camelCase

Co-authored-by: Ajinkya Kamat <kamat.ajinkya@gmail.com>

aayush-fadia committed:
Remove debugging toast

Co-authored-by: Ajinkya Kamat <kamat.ajinkya@gmail.com>

aayush-fadia committed:
Make names camelCase

aayush-fadia committed:
Moved fetching of stored Robots to the a presenter.

aayush-fadia committed:
Sort Robots with less Boilerplate.

aayush-fadia commented:
Moved the Robot List fetcher to the MainActivity Presenter.

aayush-fadia committed:
User can now add Robots, which are persisted across restarts.
Added robots instantly appear in MainActivity.
Added a model for handling robot storage, and a few other misc changes to the Main activity and presenter.

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainActivity.kt:
```diff
@@ -1,12 +1,82 @@
 package `in`.ivlabs.shayak.view
```
Move file to MainActivityFolder

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainActivityPresenter.kt:
```diff
@@ -0,0 +1,40 @@
+package `in`.ivlabs.shayak.view
```
Move to MainActivity Folder

kamatajinkya commented on app/src/main/java/in/ivlabs/shayak/view/MainBottomSheet.kt:
```diff
@@ -0,0 +1,45 @@
+package `in`.ivlabs.shayak.view
```
Move file to mainActivityFolder

kamatajinkya reviewed (changes requested):
Looks good there is a small change after which it is good to be merged

aayush-fadia committed:
Moved files to mainactivity folder

aayush-fadia committed:
Connects to WiFi in Android 10.
Debugging sending a simple string to a Server

aayush-fadia committed:
Robot now connects to the wifi, and can send string messages to a tcp server on it.

aayush-fadia committed:
Added code from navigateactivity branch here
Will begin to work on RTSP for streaming video.
